documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -0,0 +1,473 @@
+<3> Add a <<search-dsl-projection-field,regular field projection>> on a "title" field.
+<4> Add a <<search-dsl-projection-highlight,highlight projection>> on a "description" field.
+<5> Provide a predicate to filter documents and to use to highlight the results.
+<6> Fetch the results.
+====
+
+A highlighter behavior can be configured. See various available <<search-dsl-highlighting-highlighter-options,configuration options>>.
+A highlighter definition is provided after a where clause of a query:
+
+[[search-dsl-highlighting-highlighter-confguring-default-highlighter-simple]]
+.Configuring a default highlighter
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=very-basic-config]
+----
+<1> Specify a predicate that would look for matches in both `title` and `description` fields.
+<2> Specify the details of the default highlighter. Setting the no match size to a positive value to let the highlighter know
+that we want to get some text back even if there will be nothing to highlight in a particular field.
+<3> Fetch the results.
+====
+
+[[search-dsl-highlighting-highlighter-type]]
+== Highlighter type
+
+Before a highlighter can be configured, you need to pick its type:
+
+Picking the highlighter type is the first step in a highlighter definition:
+
+.Specifying the highlighter type
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=various-highlighter-types]
+----
+<1> Starting the definition of a plain highlighter.
+<2> Starting the definition of a unified highlighter.
+<3> Starting the definition of a fast vector highlighter.
+====
+
+There are three options to choose from when it comes to the highlighter type:
+
+[[search-dsl-highlighting-highlighter-type-plain]]
+Plain::
+The plain highlighter can be useful for simple queries targeting a single field on a small number of documents.
+This highlighter uses a standard Lucene highlighter. It reads the string value of a highlighted field,
+then creates a small in-memory index from it and applies query logic to perform the highlighting.
+[[search-dsl-highlighting-highlighter-type-unified]]
+Unified::
+The unified highlighter is used by default and does not necessarily rely on re-analyzing the text, as it can
+get the offsets either from postings or from term vectors.
++
+This highlighter uses a break iterator (breaks the text into sentences by default) to break the text into later scored passages.
+It better supports more complex queries. Since it can work with prebuilt data, it performs better in case of a larger amount
+of documents compared to the plain highlighter.
+[[search-dsl-highlighting-highlighter-type-fast-vector]]
+Fast vector::
+In addition to using a break iterator similar to the unified highlighter, it can use the boundary characters
```

Comment:
It's a bit confusing to use "it" to refer to something that you didn't really name in a previous sentence.

```suggestion
The fast vector highlighter, in addition to using a break iterator similar to the unified highlighter, can use the boundary characters
```